mods/cnc/maps/gdi08b/gdi08b-AI.lua

```diff
@@ -0,0 +1,249 @@
+	local toBuild = { Utils.Random(InfantryProductionTypes) }
+	local Path = Utils.Random(AttackPaths)
+	building.Build(toBuild, function(unit)
+		InfantryAttackGroup[#InfantryAttackGroup + 1] = unit[1]
+
+		if #InfantryAttackGroup >= InfantryGroupSize then
+			MoveAndHunt(InfantryAttackGroup, Path)
+			InfantryAttackGroup = { }
+			Trigger.AfterDelay(InfantryProductionCooldown, function() ProduceInfantry(building) end)
+		else
+			Trigger.AfterDelay(delay, function() ProduceInfantry(building) end)
+		end
+	end)
+end
+
+ProduceVehicle = function(building)
+	if building.IsDead or building.Owner ~= Nod then
+		return
+	elseif not CheckForHarvester() then
+		ProduceHarvester(building)
+		Trigger.AfterDelay(DateTime.Seconds(10), function() ProduceVehicle(building) end)
+		return
+	end
+
+	local delay = Utils.RandomInteger(DateTime.Seconds(12), DateTime.Seconds(17))
+	local toBuild = { Utils.Random(VehicleProductionTypes) }
+	local Path = Utils.Random(AttackPaths)
+	building.Build(toBuild, function(unit)
+		VehicleAttackGroup[#VehicleAttackGroup + 1] = unit[1]
+
+		if #VehicleAttackGroup >= VehicleGroupSize then
+			MoveAndHunt(VehicleAttackGroup, Path)
+			VehicleAttackGroup = { }
+			Trigger.AfterDelay(VehicleProductionCooldown, function() ProduceVehicle(building) end)
+		else
+			Trigger.AfterDelay(delay, function() ProduceVehicle(building) end)
+		end
+	end)
+end
+
+StartAI = function()
+	Nod.Cash = StartingCash
+	--GuardBase()
+end
+
+Trigger.OnAllKilledOrCaptured(NodBase, function()
+	Utils.Do(Nod.GetGroundAttackers(), IdleHunt)
+end)
+
+Trigger.OnKilled(nodrefinery, function()
+	BuildBuilding(BaseRefinery, NodYard)
+end)
+
+Trigger.OnKilled(nodradar, function()
+	BuildBuilding(BaseRadar, NodYard)
+end)
+
+Trigger.OnKilled(nodpower1, function()
+	BuildBuilding(BaseNuke1, NodYard)
+end)
+
+Trigger.OnKilled(nodpower2, function()
+	BuildBuilding(BaseNuke2, NodYard)
+end)
+
+Trigger.OnKilled(nodpower3, function()
+	BuildBuilding(BaseNuke3, NodYard)
+end)
+
+Trigger.OnKilled(nodpower4, function()
+	BuildBuilding(BaseNuke4, NodYard)
+end)
+
+Trigger.OnKilled(nodpower5, function()
+	BuildBuilding(BaseNuke5, NodYard)
+end)
+
+Trigger.OnKilled(gun1, function()
+	BuildBuilding(BaseGun1, NodYard)
+end)
+
+Trigger.OnKilled(gun2, function()
+	BuildBuilding(BaseGun2, NodYard)
+end)
+
+Trigger.OnKilled(gun3, function()
+	BuildBuilding(BaseGun3, NodYard)
+end)
+
+Trigger.OnKilled(gun4, function()
+	BuildBuilding(BaseGun4, NodYard)
+end)
+
+Trigger.OnKilled(nodsilo1, function()
+	BuildBuilding(BaseSilo1, NodYard)
+end)
+
+Trigger.OnKilled(nodsilo2, function()
+	BuildBuilding(BaseSilo2, NodYard)
+end)
+
+Trigger.OnKilled(nodsilo3, function()
+	BuildBuilding(BaseSilo3, NodYard)
+end)
+
+Trigger.OnKilled(handofnod, function()
+	BuildBuilding(InfantryProduction, NodYard)
+end)
+
+Trigger.OnKilled(nodairfield, function()
+	BuildBuilding(VehicleProduction, NodYard)
+end)
+
+-- units destroyed, send heli, eg. harv, tnk, bggy,...
+Trigger.OnKilled(harv1, function()
```

Comment:
Should be able to combine all these into a table and use
```
Utils.Do(NodUnits, function(unit)
	Trigger.OnKilled(unit, function()
		SendHeli(NodHelis[2])
```

Reply:
Before I update the PR, I've been wondering: How does lua count? Is `NodHelis[2]` correct (1,2,3), or should it be `NodHelis[1]` (0,1,2)?

Reply:
It counts 1,2,3